ops/charm.py

```diff
@@ -826,6 +826,79 @@ def defer(self) -> None:
             'this event until you create a new revision.')
 
 
+class CollectStatusEvent(EventBase):
+    """Event triggered at the end of every hook to collect statuses for evaluation.
+
+    If the charm wants to provide application or unit status in a consistent
+    way after the end of every hook, it should observe the
+    :attr:`collect_app_status <CharmEvents.collect_app_status>` or
+    :attr:`collect_unit_status <CharmEvents.collect_unit_status>` event,
+    respectively.
+
+    The framework will trigger these events after the hook code runs
+    successfully (``collect_app_status`` will only be triggered on the leader
+    unit). If any statuses were added by the event handlers using
+    :meth:`add_status`, the framework will choose the highest-priority status
+    and set that as the status (application status for ``collect_app_status``,
+    or unit status for ``collect_unit_status``).
+
+    The order of priorities is as follows, from highest to lowest:
+
+    * error
+    * blocked
+    * waiting
```

Comment:
Good call -- updated.

Reply:
For the record, I flipped this back, because Juju itself has waiting > maintenance, and I want to be consistent with what Juju uses: https://github.com/juju/juju/blob/3ce6ae0d106659251619f0757b569b57cb71217d/core/status/status.go#L373-L381